triples-generator/src/main/scala/io/renku/triplesgenerator/events/consumers/tsmigrationrequest/migrations/MigrationToNamedGraphs.scala

```diff
@@ -0,0 +1,175 @@
+import io.renku.triplesstore._
+import org.typelevel.log4cats.Logger
+import tooling.{RecordsFinder, RecoverableErrorsRecovery}
+
+private class MigrationToNamedGraphs[F[_]: Async](
+    defaultGraphProjectsFinder: DefaultGraphProjectsFinder[F],
+    namedGraphsProjectFinder:   NamedGraphsProjectFinder[F],
+    eventsSender:               EventSender[F],
+    recoveryStrategy:           RecoverableErrorsRecovery = RecoverableErrorsRecovery
+) extends Migration[F] {
+
+  import defaultGraphProjectsFinder._
+  import eventsSender._
+  import fs2._
+  import namedGraphsProjectFinder._
+  import recoveryStrategy._
+
+  override lazy val name: Migration.Name = MigrationToNamedGraphs.name
+
+  override def run(): EitherT[F, ProcessingRecoverableError, Unit] = EitherT {
+    Stream
+      .iterate(1)(_ + 1)
+      .evalMap(findDefaultGraphProjects)
+      .takeThrough(_.nonEmpty)
+      .flatMap(in => Stream.emits(in))
+      .evalFilterNot { case (projectId, _) => checkExists(projectId) }
+      .map(toRedoEvent)
+      .evalMap { case (payload, ctx) => sendEvent(payload, ctx) }
+      .compile
+      .drain
+      .map(_.asRight[ProcessingRecoverableError])
+      .recoverWith(maybeRecoverableError[F, Unit])
+  }
+
+  private lazy val toRedoEvent: ProjectInfo => (EventRequestContent.NoPayload, EventSender.EventContext) = {
+    case (_, path) =>
+      val eventCategoryName = CategoryName("EVENTS_STATUS_CHANGE")
+      EventRequestContent.NoPayload {
+        json"""{
+          "categoryName": $eventCategoryName,
+          "project": {
+            "path": $path
+          },
+          "newStatus": $TriplesGenerated
+        }"""
+      } -> EventSender.EventContext(eventCategoryName, show"$categoryName: $name cannot send event for $path")
+  }
+}
+
+private object MigrationToNamedGraphs {
+  val name: Migration.Name = Migration.Name("Migration to Named Graphs")
+
+  def apply[F[_]: Async: Logger: MetricsRegistry: SparqlQueryTimeRecorder] = for {
+    defaultGraphProjectsFinder <- DefaultGraphProjectsFinder[F]
+    namedGraphsProjectFinder   <- NamedGraphsProjectFinder[F]
+    eventsSender               <- EventSender[F]
+  } yield new MigrationToNamedGraphs(defaultGraphProjectsFinder, namedGraphsProjectFinder, eventsSender)
+}
+
+private trait DefaultGraphProjectsFinder[F[_]] {
+  def findDefaultGraphProjects(page: Int): F[List[ProjectInfo]]
+}
+
+private object DefaultGraphProjectsFinder {
+
+  type ProjectInfo = (projects.ResourceId, projects.Path)
+
+  def apply[F[_]: Async: Logger: SparqlQueryTimeRecorder]: F[DefaultGraphProjectsFinder[F]] =
+    RenkuConnectionConfig[F]().map(RecordsFinder[F](_)).map(new DefaultGraphProjectsFinderImpl(_))
+}
+
+private class DefaultGraphProjectsFinderImpl[F[_]](recordsFinder: RecordsFinder[F])
+    extends DefaultGraphProjectsFinder[F]
+    with Schemas {
+
+  import ResultsDecoder._
+  import io.renku.tinytypes.json.TinyTypeDecoders._
+
+  private val chunkSize: Int = 30
```

Comment:
Do you think we could increase this? It's "only" id and path what is loaded, right?

Reply:
Yes, we could. So far the value is absolutely arbitrary and simply on a very, very defensive side.

Reply:
Raised it to 50 :)